canvas/models/models.py

```diff
@@ -175,9 +176,9 @@ def username(self):
 
     @property
     def name(self):
-        if self.user.has_name:
-            return self.user.get_full_name()
-        return "Anonymous Student"
+        if self.user.has_nickname:
+            return self.user.nickname
+        return generate_default_name()
```

Comment:
If you're doing this because the nickname could be an empty string, thing returning a freshly generated nickname would be a bad idea, no?

It would be different every time and it's not saving this newly generated default nickname to the database